ls1intum/Artemis
# `Programming exercises`: Show test case details consistently for tutors on all related pages

### Checklist
#### General
#### Client

### Motivation and Context
See #5959

When an editor or instructor is working with automatic feedback on Artemis, there currently is not consistent behavior as to whether or not the test case names are shown to the user. For example, when viewing a student's score via the Scores-Page, the test case names are shown, but when using the submissions-page instead, no more names are present.

### Description
This PR aims to create a more unified behavior. Instead of allowing every component to pass a boolean from the outside, now the feedback component itself determines if test case names should be shown.

Test case names now get consistently shown if either: 
- The programming exercise has the option "show test case names to students" active
- The user is a least an Editor in the course

Test all places where a PE result is shown:
- The result view on the top right of the exercise
- Clicking on a task name in the problem statement
- Clicking on "show more results"
- Clicking on the task icons on the top of the page
- While using the online code editor 

For editors, also check the following places (names should always be show)
- Scores table
- Submissions Table
- Individual Submissions Table (Click on the number of submissions on the scores table)
- Template / Solution Result
- Template / Solution Submission (Click on "Show Submissions" under the result)
- Assessment Dashboard (for exercises with manual correction)
- Assessment Code Editor (for exercises with manual correction)

### Steps for Testing
Prerequisites:
- 2 Programming Exercises ("Show test case names to students off and on")
- 1 Tutor
- 1 Student

Check that the student only sees test case names if the option is set on.

#### Exam Mode Testing
Similar to above, test that students cannot set test case names while solving an exam exercise or when reviewing the exam results.

### Review Progress

#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
all changed covered

### Screenshots
